app/controllers/proposals_controller.rb

```diff
@@ -49,10 +49,12 @@ def approve
   end
 
   def complete
-    step = proposal.existing_or_delegated_step_for(current_user)
-    step.update_attributes!(completer: current_user)
-    step.complete!
-    flash[:success] = "You have approved #{proposal.public_id}."
+    step = proposal.existing_or_delegated_actionable_step_for(current_user)
+    if step
+      complete_step(step)
+    else
+      flash[:alert] = "There are no actionable steps available for you."
```

Comment:
doesn't that mean that we should never git this case in the controller in real life? my instinct is that if we can't construct a test to mimic this situation, we can't mimic IRL either? or am I missing something?